### PR TITLE
OCSADV-373: Take GMOS read mode into account for ITC warnings.

### PR DESCRIPTION
**Pull request by fnussber**
Minor change to make sure that ITC takes GMOS read modes for the gain calculations into account for saturation warnings. 

As part of this PR combo boxes for amp gain and read mode have been added to the GMOS web pages (see below) in order to bring everything in sync.

image

### CODE REVIEW

**Comment by andrewwstephens** [comment]
(thumbsup)

**Comment by andrewwstephens** [comment]
+1

**Comment by andrewwstephens** [comment]
Last try:  :+1:

File: bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gmos/GmosRecipe.java [hunk 1]
```diff
@@ -301,14 +305,18 @@ private ImagingResult calculateImagingDo(final Gmos instrument) {
 
     // TODO: some of these warnings are similar for different instruments and could be calculated in a central place
     private List<ItcWarning> warningsForImaging(final Gmos instrument, final double peakPixelCount) {
-        final double wellLimit     = 0.95 * instrument.getWellDepth() * instrument.getSpatialBinning() * instrument.getSpectralBinning();
-        final double lowGainLimit  = 0.95 * instrument.getADSaturation() * instrument.getLowGain();
-        final double highGainLimit = 0.95 * instrument.getADSaturation() * instrument.getHighGain();
+        final double wellLimit = 0.95 * instrument.getWellDepth() * instrument.getSpatialBinning() * instrument.getSpectralBinning();
+        final double meanGain;
+        switch (gmosParameters.site()) {
+            case GN: meanGain = InstGmosNorth.getMeanGain(gmosParameters.ampGain(), gmosParameters.ampReadMode(), gmosParameters.ccdType()); break;
```
Comment (cquiroz) [inline bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gmos/GmosRecipe.java]: isn't `getMeanGain` overloaded on each instrument?

File: bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gmos/InstGmosCommon.java [hunk 2]
```diff
@@ -489,7 +485,7 @@ private void _setADC(String name) {
      * <p/>
      * The E2V CCD's are being replaced in GMOS-B by Hamamatsu CCDs
      */
-    public String getActualGain() {
+    public double getActualGain() {
```
Comment (cquiroz) [inline bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gmos/InstGmosCommon.java]: :+1:

**Comment by cquiroz** [comment]
It s is certainly an improvement that you return `double` and `int` instead of `String` with numbers inside :smile: 

Looks good :+1:

File: bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gmos/GmosRecipe.java (hunk 1 above)
Comment (fnussber) [inline bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gmos/GmosRecipe.java]: It is, the problem is I don't have an spModel GMOS instrument object available here. I could create one, but instead I turned that method into a static method so I can use it from the ITC code without instantiating an spModel GMOS instrument.

(Ideally at some point we can combine the ITC instruments with the spModel ones, but I think that's going to be a long way to go..)

**Comment by fnussber** [comment]
Yeah, baby steps, baby steps... :smile:

**Pull request merged**